### ISSUE TITLE
Using the "mps" device on x86 Mac with AMD gpu, torch.argmax returns incorrect results.

### ISSUE THREAD

**Issue by billvsme** [pytorch/pytorch#112186]
### 🐛 Describe the bug

When using the **"mps"** device on **x86 mac** with **AMD** gpu, **torch.argmax** returns incorrect results. 
For example, if I use CPU, the results are correct. However, when I use the "mps" device, the results are incorrect, all of which are 0.

```
>>> import torch
>>> a = torch.tensor([[0, 0, 1], [0, 1, 0], [1, 0, 0]], dtype=torch.int)
>>> a.argmax(dim=1)
tensor([2, 1, 0])
>>> a = a.to("mps")
>>> a.argmax(dim=1)
tensor([0, 0, 0], device='mps:0')
>>> torch.__version__
'2.1.0'
```

Guessing it is due to certain differences between M1 and X86 mac or AMD gpu.

### Versions

PyTorch version: 2.1.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 12.7 (x86_64)
GCC version: Could not collect
Clang version: 14.0.0 (clang-1400.0.29.202)
CMake version: version 3.17.2
Libc version: N/A

Python version: 3.9.13 (main, May 24 2022, 21:28:12)  [Clang 12.0.0 (clang-1200.0.32.29)] (64-bit runtime)
Python platform: macOS-10.16-x86_64-i386-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz

Versions of relevant libraries:
[pip3] mypy-extensions==1.0.0
[pip3] numpy==1.23.5
[pip3] onnxruntime==1.16.1
[pip3] torch==2.1.0
[pip3] torchvision==0.16.0
[conda] Could not collect

cc @kulinseth @albanD @malfet @DenisVieriu97 @razarmehr @abhudev

**Comment by booysej** [comment]
Have more or less the same MacBook Pro A1706  15-inch, Late 2016 model. In this configuration, the MacBook Pro features dual graphics cards: the Intel HD Graphics 530, which is an integrated GPU, and an additional discrete AMD Radeon Pro GPU. used the latest torch 2.3.0.dev20240110 but also getting the same.

Python 3.10.13 | packaged by conda-forge | (main, Dec 23 2023, 15:36:59) [Clang 16.0.6 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import torch
>>> a = torch.tensor([[0, 0, 1], [0, 1, 0], [1, 0, 0]], dtype=torch.int)
>>> a.argmax(dim=1)
tensor([2, 1, 0])
>>> a = a.to("mps")
>>> a.argmax(dim=1)
tensor([0, 0, 0], device='mps:0')
>>> torch.__version__
'2.3.0.dev20240110'
>>>

**Comment by skotapati** [comment]
Hello, x86 issues are currently not a high priority for further investigation. If ops have correctness issues, it's recommended to fall back to CPU execution when possible
